src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -6744,14 +6738,14 @@ private static void loopChecks1a(int i, MethodHandle in, MethodHandle st) {
     }
 
     private static List<Class<?>> longestParameterList(Stream<MethodHandle> mhs, int skipSize) {
-        return mhs.filter(Objects::nonNull).
+        return mhs.filter(Objects::nonNull)
                 // take only those that can contribute to a common suffix because they are longer than the prefix
-                        map(MethodHandle::type).
-                        filter(t -> t.parameterCount() > skipSize).
-                        map(MethodType::ptypes).
-                        reduce((p, q) -> p.length >= q.length ? p : q).
-                        map(longest -> List.of(Arrays.copyOfRange(longest, skipSize, longest.length))).
-                        orElse(List.of());
+                .map(MethodHandle::type)
+                .filter(t -> t.parameterCount() > skipSize)
+                .max(Comparator.comparing(MethodType::parameterCount))
+                .map(MethodType::ptypes)
+                .map(longest -> List.of(Arrays.copyOfRange(longest, skipSize, longest.length)))
```

Comment:
i think you can fuse these to map() calls

Reply:
Done!